exporter/elasticsearchexporter/exporter_test.go

```diff
@@ -216,7 +217,10 @@ func TestExporterLogs(t *testing.T) {
 		server := newESTestServer(t, func(docs []itemRequest) ([]itemResponse, error) {
 			rec.Record(docs)
 
-			assert.Equal(t, "logs-record.dataset-resource.namespace", actionJSONToIndex(t, docs[0].Action))
+			ds := "record.dataset" + strings.Repeat("_", len(disallowedDatasetRunes))
```

Comment:
I'm a little hesitant about only updating this test in exporter_test. We should either update the equiv test for logs, metrics, and traces, or do not update at all.